src/FINDOLOGIC/Api/RequestBuilders/Json/SuggestionRequestBuilder.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace FINDOLOGIC\Api\RequestBuilders\Json;
+
+use FINDOLOGIC\Api\Config;
+use FINDOLOGIC\Api\Definitions\Endpoint;
+use FINDOLOGIC\Api\Definitions\QueryParameter;
+use FINDOLOGIC\Api\Exceptions\InvalidParamException;
+use FINDOLOGIC\Api\RequestBuilders\RequestBuilder;
+use FINDOLOGIC\Api\ResponseObjects\Json\JsonResponse;
+use FINDOLOGIC\Api\Validators\ParameterValidator;
+
+class SuggestionRequestBuilder extends RequestBuilder
+{
+    protected $endpoint = Endpoint::SUGGESTION;
+
+    public function __construct(Config $config)
+    {
+        parent::__construct($config);
+        $this->addRequiredParam(QueryParameter::QUERY);
+    }
+
+    /**
+     * @inheritdoc
+     * @return JsonResponse
+     */
+    public function sendRequest()
+    {
+        $this->checkRequiredParamsAreSet();
+
+        $responseContent = $this->client->request($this->buildRequestUrl());
+        return new JsonResponse($responseContent, $this->client->getResponseTime());
+    }
+
+    /**
+     * Sets the query param. It is used as the search query. When doing a suggestion request the min length for a
+     * query is one character.
+     *
+     * @param $value string
+     * @see https://docs.findologic.com/doku.php?id=smart_suggest_new#request
+     * @return $this
+     */
+    public function setQuery($value)
+    {
+        $validator = new ParameterValidator([QueryParameter::QUERY => $value]);
+        $validator
+            ->rule('string', QueryParameter::QUERY)
+            ->rule('lengthMin', QueryParameter::QUERY, 1);
+
+        if (!$validator->validate()) {
+            throw new InvalidParamException(QueryParameter::QUERY);
+        }
+
+        $this->addParam(QueryParameter::QUERY, $value);
+        return $this;
+    }
+
+    /**
+     * This function exists for legacy reasons. Using this function will just call setQuery instead.
```

Comment:
>This function exists for legacy reasons.

The sooner we get rid of it in anything new we make, the sooner we can deprecate and remove this parameter.